### ISSUE TITLE
Refactor Nightwatch Specs into Cypress

### ISSUE THREAD

**Issue by pjhill** [department-of-veterans-affairs/va.gov-team#27067]
# Description
The first scan of `vets-website` for Nightwatch specs may not have located them all. Upon attempting to remove the form tester helper files, it turns out there are still some specs that depend on it.  We need to look for more Nightwatch specs and refactor them into Cypress.

## Tasks
- [x] Generate a list of Nightwatch specs that are still in the repository and do not have replacement Cypress specs written

**Comment by CBonade** [comment]
edu-benefits/tests/feedback-tool/00.feedback-tool.e2e.spec.js
edu-benefits/tests/feedback-tool/01.feedback-tool.-keyboard.e2e.spec.js
pre-need/tests/00-all-fields.e2e.skip.spec.js
veteran-representative/tests/00.veteran-representative-entry.e2e.spec.js
coronavirus-research/tests/00.covid-vaccine-trial.e2e.spec.js
gi/tests/2e2/01-keyboard.e2e.spec.js
post-911-gib-status/tests/01-authed.e2e.spec.js
static-pages/tests/application-status.e2e.spec.js
terms-and-conditions/00.terms-and-conditions.e2e.spec.js
veteran-representative/tests/00.veteran-representative-entry.e2e.spec.js
forms/tests/save-in-progress/01-sip-autosave.e2e.spec.js
forms/tests/save-in-progress/01-sip-finish-later.e2e.spec.js
forms/tests/save-in-progress/01-sip-load-fail.e2e.spec.js
forms/tests/save-in-progress/01-sip-review.e2e.spec.js
platform/monitoring/DowntimeNotification/tests/DowntimeNotification.e2e.spec.js
platform/site-wide/banners/index.e2e.spec.js
platform/site-wide/tests/sanity-check/smoke-test.e2e.spec.js
platform/site-wide/user-nav/tests/user-nav.e2e.spec.js
platform/user/tests/unauthed-flows.e2e.spec.js

**Comment by CBonade** [comment]
A few of these specs, aside from the one checked, are either partially converted or have been analyzed and found to be being covered by another test. More to come in the coming days.

**Comment by CBonade** [comment]
Comments on a few of these specs (will continue to add to this list as I go):

coronavirus-research/tests/00.covid-vaccine-trial.e2e.spec.js - covered by a new spec `reviewAndSubmit.cypress.spec.js`

**Comment by CBonade** [comment]
Temporarily on hold prioritizing Cypress reliability for GHA launch

**Comment by CBonade** [comment]
Resumed working through a few of these specs today while waiting for Github actions runs to complete.  Checklist of current progress is up to date.

**Comment by CBonade** [comment]
Five more specs checked off today.  Two of the remaining specs are pending Noah's in-progress effort on keyboard testing in Cypress.  The last one, is quite a large set of data and I will begin working through that.

**Comment by CBonade** [comment]
Pre-needs spec is about 80% complete.

**Comment by CBonade** [comment]
I've removed the User nav test from the checklist as it's been problematic trying to get a cross-origin check working in Cypress. This test works locally with `chromeWebSecurity` disabled, but will not in CI.  Therefore, I've commented out those lines in the meantime to get the rest of the completed specs into our regular testing routine to start testing their stability.  As soon as the initial PR has passed checks, I'll flag it for review.  Nightwatch has been added back into it as well, while we test out these new specs.

**Comment by CBonade** [comment]
Done a fair bit of research at this point for options to test cross-origin in Cypress.  Cypress offers some workarounds, but none that work for the structure of our product. Therefore, that leaves us with just a couple options:

1) Test that the sign out button exists, assert it's text is correct, leave it at that (does not test our SSO sign out in any fashion).
2) Test that the errors we get are the correct errors for what we would expect if cross-origin was accessed in a Cypress spec, to verify at a minimum that an attempt was made to navigate to a different domain.
3) Rework the React code to either insert the href into the button which can then be verified by the test, or emit an event that can be caught and checked in the test as well that contains the url as a payload.

I went with #2 for now, and have it sitting in a branch.  It checks both that we get a network error (expected making a request from a cross-domain url) and additionally checks that we get an appropriate Chrome Error inside Cypress when attempting to navigate cross-origin.  Both these checks pass on the test:

https://github.com/department-of-veterans-affairs/vets-website/pull/18041

![image.png](https://images.zenhubusercontent.com/60aa993146d844d86f494162/d82b3bc3-d84d-44a5-a805-22f4d3ab87c2)

**Comment by CBonade** [comment]
Got the two outstanding reviews today, however unfortunately, some of the comments from that review resulted in an additional need for 7 more code owner reviews.  The biggest change made here was the removal of `{timeout: Timeouts.normal}` from all cypress specs in favor of leaning on Cypress' built-in timeout.  Other changes include moving specs around, deleting specs, among others.  I imagine this will be in review for a bit longer.

**Comment by CBonade** [comment]
One review down, five to go!

**Comment by CBonade** [comment]
User nav test issues were solved today and merged in.  The large PR with the bulk of the Nightwatch rewrites has not had any more reviews yet (still waiting on 4)

**Comment by CBonade** [comment]
No additional reviews today - still awaiting 4 teams.

**Comment by CBonade** [comment]
No new reviews, awaiting 4 more teams

**Comment by CBonade** [comment]
Only the two keyboard specs remain here, which is awaiting further development on the keyboard helper functions.  I've started a new, clean task for the keyboard test work. This one is ready to close. https://app.zenhub.com/workspaces/vsp-5cedc9cce6e3335dc5a49fc4/issues/department-of-veterans-affairs/va.gov-team/29618

**Comment by pjhill** [comment]
It's done. Why did I wait so long to close this?!
